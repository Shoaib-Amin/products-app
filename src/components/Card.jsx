import React from 'react'

function Card({ product }) {
  return (
    <div className='product_card_container'>
      <div className='product_name_container'>
        <div>
          <p className='product_name'>
            {product.name}
            <span style={{ fontWeight: 400, marginLeft: 30 }}>{`$${product.price}`}</span>
          </p>
        </div>
        <p className='date_style'>{product.inventoryDate}</p>

      </div>
      <p className='product_description'>{product.description}</p>

    </div>
  )
}

export default Card