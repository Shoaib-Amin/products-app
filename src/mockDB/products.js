export const products = [
    {
        name: 'product_1',
        description: 'description for product_1',
        price: 25,
        inventoryDate: '29-04-2023'
    },
    {
        name: 'product_2',
        description: 'description for product_2',
        price: 30,
        inventoryDate: '29-03-2023'
    },
    {
        name: 'product_3',
        description: 'description for product_3',
        price: 35,
        inventoryDate: '20-03-2023'
    },
    {
        name: 'product_4',
        description: 'description for product_4',
        price: 45,
        inventoryDate: '30-02-2023'
    },
    {
        name: 'product_5',
        description: 'description for product_5',
        price: 50,
        inventoryDate: '29-02-2023'
    },
]