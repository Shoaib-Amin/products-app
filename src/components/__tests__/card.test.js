import renderer from 'react-test-renderer';
import Card from '../Card'
import { products } from '../../mockDB/products'

test('it render the card correctly', () => {
    const cardComponent = renderer
        .create(<Card product={products[products.length -1]} />)
        .toJSON();
    expect(cardComponent).toMatchSnapshot();
})