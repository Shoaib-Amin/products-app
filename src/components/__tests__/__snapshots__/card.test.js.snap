// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the card correctly 1`] = `
<div
  className="product_card_container"
>
  <div
    className="product_name_container"
  >
    <div>
      <p
        className="product_name"
      >
        product_5
        <span
          style={
            Object {
              "fontWeight": 400,
              "marginLeft": 30,
            }
          }
        >
          $50
        </span>
      </p>
    </div>
    <p
      className="date_style"
    >
      29-02-2023
    </p>
  </div>
  <p
    className="product_description"
  >
    description for product_5
  </p>
</div>
`;
